libcaliptra/src/caliptra_api.c

```diff
@@ -348,6 +351,45 @@ bool caliptra_ready_for_firmware(void)
     return true;
 }
 
+/*
+* caliptra_is_csr_ready
+*
+* Reports if the IDEVID CSR is ready
+*
+* @return bool True if ready, false otherwise    
+*/
+bool caliptra_is_csr_ready(void)
+{
+    uint32_t status;
+
+    status = caliptra_read_status();
+
+    if ((status & GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_IDEVID_CSR_READY_MASK) == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_IDEVID_CSR_READY_MASK)
+    {
+        return true;
+    }
+ 
+    return false;
+}
+
+/*
+* caliptra_csr_is_ready
+*
+* Waits forever for the CSR to be ready
+*
+* @return bool True if ready, false otherwise    
+*/
+void caliptra_wait_for_csr_ready(void) 
+{
+    while (true) 
+    {
+        if (caliptra_is_csr_ready()) {
+            break;
+        }
```

Comment:
can we add a timeout here ?

Reply:
I don't think libcaliptra can have any platform-agnostic way of supporting timing. If the caller wants a timeout, I think they would need to directly use caliptra_is_csr_ready() in their own generic test-with-timeout function/macro.

Reply:
I removed this function from the public api and left it as test fixture.